### ISSUE TITLE
foreman_host.name[index].interfaces_attributes is empty list of object

### ISSUE THREAD

**Issue by togoschi** [terraform-coop/terraform-provider-foreman#108]
With v0.5.x the interface_attribute[index].ip attribute of all host resources are emtpy when running terraform apply for the first time.
If you run tf apply for the second time after all foreman_host resources are created the issue disappears.

An example output when `foreman_host.osd[0]` resource only is applied (and `foreman_host.osd[1]` already exists)

```
╷
│ Error: Invalid index
│ 
│   on variables.tf line 141, in locals:
│  141:   mgr_ips = formatlist("%s", concat(tolist([foreman_host.osd[0].interfaces_attributes[0].ip, foreman_host.osd[1].interfaces_attributes[0].ip]))
│     ├────────────────
│     │ foreman_host.osd[0].interfaces_attributes is empty list of object
│ 
│ The given key does not identify an element in this collection value.
╵
time=2023-01-11T17:57:05+01:00 level=error msg=1 error occurred:
        * exit status 1
```

Terraform Code

```
locals {
  mgr_ips = formatlist("%s", tolist([foreman_host.osd[0].interfaces_attributes[0].ip, foreman_host.osd[1].interfaces_attributes[0].ip]))
}

resource "foreman_host" "osd" {
  count      = var.osd.count
  name       = format(var.osd.hostname_format, count.index + 1)
  build      = true
  managed    = true
  enable_bmc = false

  interfaces_attributes {
    type      = "interface"
    primary   = true
    provision = true
    managed   = true
    subnet_id = data.foreman_subnet.hcp-infra.id
    compute_attributes = {
      "network" = var.vsphere_client_network
    }
  }

  interfaces_attributes {
    type      = "interface"
    primary   = false
    provision = false
    managed   = true
    compute_attributes = {
      "network" = var.vsphere_cluster_network
    }
  }
```

**Comment by agriffit79** [comment]
I know what the problem is. I am working on a fix.

**Comment by thatsk** [comment]
this is still issue i just checked its not working

**Comment by agriffit79** [comment]
I was able to re-create the issue and the fix resolved for me.

@togoschi - were you able to test it?

**Comment by togoschi** [comment]
@agriffit79 Yes, the issue is fixed in v0.5.8
